File: samples/modular-kmp-ui-data-layer/androidApp/src/main/kotlin/com/example/kmpsamples/modularkmpuidatalayer/android/DetailsViewModel.kt
package com.example.kmpsamples.modularkmpuidatalayer.android

import com.example.kmpsamples.modularkmpuidatalayer.featuretwoshareddata.DetailsRepository
import com.example.kmpsamples.modularkmpuidatalayer.featuretwosharedui.DetailsUiState

class DetailsViewModel(
    private val repository: DetailsRepository = DetailsRepository()
) {
    fun load(platform: String): DetailsUiState = DetailsUiState(
        title = "Native DetailsViewModel on $platform",
        body = repository.loadDetailsSummary(platform)
    )

    fun refresh(platform: String): DetailsUiState = DetailsUiState(
        title = "Native DetailsViewModel refreshed on $platform",
        body = repository.loadDetailsSummary(platform)
    )
}

File: samples/13_modular-kmp-ui-data-layer/iosApp/Sources/DetailsViewModel.swift
import FeatureTwoSharedUI
import FeatureTwoSharedData

final class DetailsViewModel {
    private let repository = DetailsRepository()

    func load(platform: String) -> DetailsUiState {
        DetailsUiState(title: "Native DetailsViewModel on \(platform)", body: repository.loadDetailsSummary(platform: platform))
    }

    func refresh(platform: String) -> DetailsUiState {
        DetailsUiState(title: "Native DetailsViewModel refreshed on \(platform)", body: repository.loadDetailsSummary(platform: platform))
    }
}
